### PR TITLE
Remove out.sh script

### PR DESCRIPTION
**Pull request by Hamms**
This one will probably be easiest to review commit-by-commit.

Overall, it moves **most** of the functionality of the out.sh script into the out.rb script, rather than calling the former from the latter.

The one major change it makes to the out.sh script is to no longer call the `fix-ruby-yml.pl` script, which I have confirmed we don't actually need for the "out" sync. 

Next steps (in future PRs):
- Verify that we don't need `fix-ruby-yml.pl` for `sync-pegasus.sh` and `in.sh`, either, and remove it entirely.
- Figure out if we still need to special-case Armenian or not
- merge `codeorg-messages.sh` functionality in here, too, so we can also remove that dependency

### CODE REVIEW

**Comment by Hamms** [comment]
[Tech debt: Improvements to “out” sync scripts](https://trello.com/c/Hvs7wp8Z/126-tech-debt-improvements-to-out-sync-scripts)

**Comment by joshlory** [comment]
Diff with whitespace changes ignored: https://github.com/code-dot-org/code-dot-org/pull/22727/files?w=1

**Comment by joshlory** [comment]
Are you looking for feedback on the code moved to `sync-codeorg-out.rb`, or just a review that the functionality hasn't changed?

File: bin/i18n/sync-codeorg-out.rb [hunk 1]
```diff
@@ -31,6 +34,80 @@ def rename_from_crowdin_name_to_locale
   end
 end
 
+# Distribute downloaded translations from i18n/locales
+# back to blockly, apps, pegasus, and dashboard.
+def distribute_translations
```
Comment (joshlory) [inline bin/i18n/sync-codeorg-out.rb:40]: Would be good to split up this huge method.

The Dashboard, Apps, Blockly Core, and Pegasus steps have similar structure — can we extract a common helper method?

File: bin/i18n/sync-codeorg-out.rb [hunk 2]
```diff
@@ -31,6 +34,80 @@ def rename_from_crowdin_name_to_locale
...
+
+    # Merge YML file.
+    if locale == 'hy-AM' # Armenian accepts English translations, does not need fallback
+      export_without_merge "yml", loc_file, destination
+    else
+      merge_translation "yml", source, loc_file, destination
+    end
+  end
+
+  puts "Distribution finished!      "
```
Comment (joshlory) [inline bin/i18n/sync-codeorg-out.rb]: Trailing spaces in this print line?

File: bin/i18n/sync-codeorg-out.rb [hunk 3]
```diff
@@ -31,6 +34,80 @@ def rename_from_crowdin_name_to_locale
...
+
+      if locale == 'hy-AM' # Armenian accepts English translations, does not need fallback
+        export_without_merge "yml", loc_file, destination
+      else
+        merge_translation "yml", source, loc_file, destination
+      end
+    end
+
+    ### Apps
+    js_locale = locale.tr('-', '_').downcase
```
Comment (joshlory) [inline bin/i18n/sync-codeorg-out.rb]: Would be great to do a one-time rename so we don't need to think about this special case.

File: bin/i18n-codeorg/lib/export-without-merge.rb [hunk 4]
```diff
@@ -3,28 +3,26 @@
...
+  # is this a new file being translated?
+  if File.exist?(new_translation_path) && !File.exist?(prev_translation_path)
+    FileUtils.copy(new_translation_path, prev_translation_path)
+  end
 
-# Translation begins
-if file_type == "yml"
-  new_translation = YAML.load_file(new_translation_path)
+  # Translation begins
+  if file_type == "yml"
```
Comment (joshlory) [inline bin/i18n-codeorg/lib/export-without-merge.rb]: Nit: prefer [single quotes for strings without interpolation](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#ruby-string-quotes).

File: bin/i18n/sync-codeorg-out.rb (hunk 2 above)
Comment (Hamms) [reply bin/i18n/sync-codeorg-out.rb]: just a lazy way to overwrite the previous line

File: bin/i18n/sync-codeorg-out.rb (hunk 3 above)
Comment (Hamms) [reply bin/i18n/sync-codeorg-out.rb]: You mean like actually move the files? That would probably be a decently large work item

File: bin/i18n/sync-codeorg-out.rb (hunk 1 above)
Comment (Hamms) [reply bin/i18n/sync-codeorg-out.rb:40]: looking over this closer, it feels like the main thing these all share in common is the special-casing for Armenian, which I'm pretty sure I'm gonna be able to remove (in a separate PR).

I vote we leave this as-is for now, and when I figure out whether or not I can stop special-casing Armenian I'll improve this then

**Comment by Hamms** [comment]
PTAL!

**Comment by Hamms** [comment]
Verified that we no longer need to special-case Armenian and removed the relevant code

**Pull request merged**